build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/rules/BinaryBreakingChangesRule.java

```diff
@@ -36,8 +36,9 @@ public class BinaryBreakingChangesRule extends AbstractGradleViolationRule {
         JApiCompatibilityChange.INTERFACE_ADDED               // the added methods will be reported
     );
 
-    public BinaryBreakingChangesRule(Map<String, String> acceptedApiChanges) {
-        super(acceptedApiChanges);
+    @SuppressWarnings("unchecked")
+    public BinaryBreakingChangesRule(Map<String, Object> params) {
+        super((Map<String, String>)params.get("acceptedApiChanges"), (String)params.get("apiChangesJsonFile"), (String)params.get("projectRootDir"));
```

Comment:
To avoid needing to duplicate all these casts and key strings, can we move the `get`s into the superclass's constructor?

Reply:
We can, it just means some adjustments are needed in the tests.